flask_potion/schema.py

```diff
@@ -254,13 +254,13 @@ def convert(self, instance, update=False, pre_resolved_properties=None, patchabl
         return result
 
     def parse_request(self, request):
-        if request.method in ('POST', 'PATCH', 'PUT', 'DELETE'):
-            if request.mimetype != 'application/json':
-                raise RequestMustBeJSON()
-
         # TODO change to request.get_json(silent=False) to catch invalid JSON
         data = request.get_json(silent=True)
 
+        if request.method in ('POST', 'PATCH', 'PUT', 'DELETE'):
+            if data and request.mimetype != 'application/json':
```

Comment:
I believe `data and request.mimetype != 'application/json'` is always `False`. `request.get_json()` will always return `None` if the mimetype is not `'application/json'` so `RequestMustBeJSON` never gets raised.

The check that should be made instead is if data is required, e.g. `self.fields` is not empty, require a JSON request otherwise allow an empty request.

```
if request.method in ('POST', 'PATCH', 'PUT', 'DELETE'):		
    if self.fields and request.mimetype != 'application/json':		
        raise RequestMustBeJSON()
```

This could be moved back into the top of the function.

A unit test should be added to make sure.